docs/zkapps/o1js/gadgets.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Gadgets
+hide_title: true
+description: How gadgets work in o1js.
+keywords:
+  - smart contracts
+  - zkapp
+  - gadgets
+  - bitwise
+  - o1js
+  - blockchain
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Gadgets
+
+Gadgets are amalgamations of the lowest level gates in the proof system (addition, multiplication) into more sophisticated boolean operations (XOR, NOT). Gadgets in o1js simplify the process of creating new cryptographic primitives and make it possible to implement other crypto algorithms, such as SHA. 
+
+<!--- I'm grabbing context from other sources SWAG, please review carefully--->
+Think of gadgets as atomic logic gates that mask large circuits. 
+
+Gadgets are small, reusable building blocks you can use to create more complex cryptographic constructions. 
+
+Leverage the properties and behavior of these smaller components to build secure and efficient systems.
+
+Questions: 
+- are all gadgets bitwise provers? do they all use ZkProgram? what does that mean? https://github.com/o1-labs/o1js/blob/main/src/examples/zkprogram/gadgets.ts#L18 
+- are all gadgets used/called as a method that moves the result into one of our 8 fields?
+- handle the constant case by using the added functionality in the bindings. For the prover case, how to describe? or go into detail for each gadget?
+- do we want to list gadgets in the gadgets namespace separately from gadgets not in the gadgets namespace? (where are these gadgets?)
+- do we have examples other than https://github.com/o1-labs/o1js/blob/main/src/examples/zkprogram/gadgets.ts? 
+- include code snippets? 
+- can we need to describe use cases? or just talk about performance gains? or?
+- primitive constraint system, we replaced Poiseidon with gadget? https://github.com/o1-labs/o1js/pull/1177/files#diff-de4ea0a8fb59a20403ab4c9ab9615587c179a8649e1b1b07297f6c829c94aa88L1 
+- resources, what do we want to provide?
+- why are gadgets listed under [variables](https://docs.minaprotocol.com/zkapps/o1js-reference/modules#variables-1) in the reference docs? is it because a gadget variable is already computed when it is used (so performance is faster?) 
```

Comment:
Technically `Gadget` itself is a variable. The reference docs are generated from the code and, since Gadget is defined as an object with function properties, the docs list `Gadget` as a variable. It's just a JavaScript thing. You'll see Poseidon, Permissions and Pickles also listed as variables. This just means that when you refer to those things they are technically defined as variables in the JS runtime. This does not need to be communicated